src/ace/managers/memory.c

```diff
@@ -164,13 +164,19 @@ void *_memAllocDbg(ULONG ulSize, ULONG ulFlags, UWORD uwLine, char *szFile) {
 #endif // AMIGA
 		return 0;
 	}
+	pAddr += sizeof(ULONG);
+	ULONG *pCafe = (ULONG*)(pAddr - sizeof(ULONG));
+	ULONG *pDead = (ULONG*)(pAddr + ulSize);
+	*pCafe = 0xCAFEBABE;
+	*pDead = 0xDEADBEEF;
```

Comment:
168-171 are useless?

Reply:
wtf mate, that's the whole point of this pull request - you set up canaries on start/end of memory alloc and during freeing it you check if it's intact